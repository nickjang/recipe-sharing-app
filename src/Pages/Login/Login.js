import React, { Component } from 'react';
import LoginForm from '../../Components/LoginForm/LoginForm';
import './Login.css';

class Login extends Component {
  handleLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/';
    history.push(destination);
    window.location.reload();
  }

  render() {
    return (
      <LoginForm onSuccess={this.handleLoginSuccess} />
    );
  }
}

Login.defaultProps = {
  location: {},
  history: {
    push: () => { }
  }
}

export default Login;