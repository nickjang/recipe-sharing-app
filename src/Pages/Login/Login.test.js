import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './Login';

describe(`Login component`, () => {
  it('renders Login by default', () => {
    const wrapper = shallow(<Login />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})