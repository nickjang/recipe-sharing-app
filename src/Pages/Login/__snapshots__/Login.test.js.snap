// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders Login by default 1`] = `
<withRouter(LoginForm)
  onSuccess={[Function]}
/>
`;
